A few dtype tests fail with `OverflowError: Python integer -1 out of bounds for uint32` on 32-bit architectures

Seen on Gentoo, both on x86 and ARM, with 0.17.0:

```pytb
=================================== FAILURES ===================================
______________________ TestConversion.test_builtin_type9 _______________________
[gw4] linux -- Python 3.12.9 /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/bin/python3.12

self = <pythran.tests.test_conversion.TestConversion testMethod=test_builtin_type9>

    def test_builtin_type9(self):
        npt = ("numpy.int8",
               "numpy.uint8",
               "numpy.int16",
               "numpy.uint16",
               "numpy.int32",
               "numpy.uint32",
               "numpy.int64",
               "numpy.uint64",
               "numpy.intp",
               "numpy.uintp",)
        for i, t in enumerate(npt):
            kwargs = {"builtin_type9p{}".format(i): [int]}
>           self.run_test("def builtin_type9p{}(x): import numpy; x = {}(x); return type(x)(x)".format(i, t),
                          1,
                          **kwargs)

i          = 5
kwargs     = {'builtin_type9p5': [<class 'int'>]}
npt        = ('numpy.int8',
 'numpy.uint8',
 'numpy.int16',
 'numpy.uint16',
 'numpy.int32',
 'numpy.uint32',
 'numpy.int64',
 'numpy.uint64',
 'numpy.intp',
 'numpy.uintp')
self       = <pythran.tests.test_conversion.TestConversion testMethod=test_builtin_type9>
t          = 'numpy.uint32'

pythran/tests/test_conversion.py:254: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:324: in run_test
    self.assertAlmostEqual(python_ref, pythran_res)
        check_exception = False
        code       = 'def builtin_type9p5(x): import numpy; x = numpy.uint32(x); return type(x)(x)'
        cxx_compiled = '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p5.cpython-312-i386-linux-gnu.so'
        interface  = {'builtin_type9p5': [<class 'int'>]}
        modname    = 'test_builtin_type9p5'
        name       = 'builtin_type9p5'
        params     = (1,)
        prelude    = None
        python_ref = np.uint32(1)
        pythran_res = np.uint32(1)
        run_pythran_args = ('test_builtin_type9p5',
 '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p5.cpython-312-i386-linux-gnu.so',
 ('builtin_type9p5', (1,)),
 None,
 False)
        self       = <pythran.tests.test_conversion.TestConversion testMethod=test_builtin_type9>
        thread_count = 1
pythran/tests/__init__.py:114: in assertAlmostEqual
    self.check_type(ref, res)
        ref        = np.uint32(1)
        res        = np.uint32(1)
        self       = <pythran.tests.test_conversion.TestConversion testMethod=test_builtin_type9>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pythran.tests.test_conversion.TestConversion testMethod=test_builtin_type9>
ref = np.uint32(1), res = np.uint32(1)

    def check_type(self, ref, res):
        """ Check if type between reference and result match. """
        print("Type of Pythran res : ", type(res))
        print("Type of Python ref : ", type(ref))
    
        # cope with subtle situation under Windows where numpy.uint32 have same
        # name but different value
        if type(res).__name__ == type(ref).__name__:
            return
    
        if isinstance(ref, ndarray):
            # res can be an ndarray of dim 0 because of isneginf call
            if ref.ndim == 0 and (not isinstance(res, ndarray)
                                  or res.ndim != 0):
                self.check_type(ref.item(0), res)
            else:
                self.assertIsInstance(res, type(ref))
        elif isinstance(ref, (float, float64)):
            self.assertIsInstance(res, (float, float64))
        elif isinstance(ref, (complex, complex128)):
            self.assertIsInstance(res, (complex, complex128))
        elif isinstance(ref, (bool, bool_)):
            self.assertIsInstance(res, (bool, bool_))
        elif hasattr(ref, 'dtype'):
            if hasattr(res, 'dtype'):
                self.assertEqual(ref.dtype.itemsize, res.dtype.itemsize)
>               self.assertEqual(ref.dtype.type(-1), res.dtype.type(-1))
E               OverflowError: Python integer -1 out of bounds for uint32

ref        = np.uint32(1)
res        = np.uint32(1)
self       = <pythran.tests.test_conversion.TestConversion testMethod=test_builtin_type9>

pythran/tests/__init__.py:102: OverflowError
----------------------------- Captured stdout call -----------------------------
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.int8'>
Type of Python ref :  <class 'numpy.int8'>
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.uint8'>
Type of Python ref :  <class 'numpy.uint8'>
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.int16'>
Type of Python ref :  <class 'numpy.int16'>
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.uint16'>
Type of Python ref :  <class 'numpy.uint16'>
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.intc'>
Type of Python ref :  <class 'numpy.int32'>
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.uintc'>
Type of Python ref :  <class 'numpy.uint32'>
----------------------------- Captured stderr call -----------------------------
INFO: Generated module: test_builtin_type9p0
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p0.cpython-312-i386-linux-gnu.so
INFO: Generated module: test_builtin_type9p1
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p1.cpython-312-i386-linux-gnu.so
INFO: Generated module: test_builtin_type9p2
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p2.cpython-312-i386-linux-gnu.so
INFO: Generated module: test_builtin_type9p3
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p3.cpython-312-i386-linux-gnu.so
INFO: Generated module: test_builtin_type9p4
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p4.cpython-312-i386-linux-gnu.so
INFO: Generated module: test_builtin_type9p5
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p5.cpython-312-i386-linux-gnu.so
------------------------------ Captured log call -------------------------------
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_builtin_type9p0' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpjesxk30l/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpxr23ismb.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpjesxk30l/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpxr23ismb.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpjesxk30l/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpxr23ismb.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp835k2sy0/test_builtin_type9p0.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_builtin_type9p0
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p0.cpython-312-i386-linux-gnu.so
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_builtin_type9p1' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpbhp7nx6w/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp12dw_ajs.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpbhp7nx6w/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp12dw_ajs.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpbhp7nx6w/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp12dw_ajs.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmppvqqk1yx/test_builtin_type9p1.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_builtin_type9p1
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p1.cpython-312-i386-linux-gnu.so
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_builtin_type9p2' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpijlo795h/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpeztv7ndz.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpijlo795h/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpeztv7ndz.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpijlo795h/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpeztv7ndz.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpz6pqvocg/test_builtin_type9p2.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_builtin_type9p2
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p2.cpython-312-i386-linux-gnu.so
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_builtin_type9p3' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmptka1d066/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpfxigmrnb.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmptka1d066/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpfxigmrnb.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmptka1d066/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpfxigmrnb.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpp61m78cq/test_builtin_type9p3.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_builtin_type9p3
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p3.cpython-312-i386-linux-gnu.so
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_builtin_type9p4' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpgvp7bt_m/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp6mtp9van.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpgvp7bt_m/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp6mtp9van.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpgvp7bt_m/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp6mtp9van.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpu30h1un_/test_builtin_type9p4.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_builtin_type9p4
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p4.cpython-312-i386-linux-gnu.so
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_builtin_type9p5' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp_cw6j5il/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp_z9gr0rf.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp_cw6j5il/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp_z9gr0rf.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp_cw6j5il/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp_z9gr0rf.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpfa7m8hwj/test_builtin_type9p5.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_builtin_type9p5
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_builtin_type9p5.cpython-312-i386-linux-gnu.so
________________________ TestNdarray.test_ndarray_uintp ________________________
[gw9] linux -- Python 3.12.9 /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/bin/python3.12

self = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>

    def test_ndarray_uintp(self):
>       self.run_test('def ndarray_uintp(a): import numpy as np; return np.uintp(a), np.array([a, a], dtype=np.uintp)',
                      numpy.uintp(5),
                      ndarray_uintp=[numpy.uintp])

self       = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>

pythran/tests/test_ndarray.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:324: in run_test
    self.assertAlmostEqual(python_ref, pythran_res)
        check_exception = False
        code       = ('def ndarray_uintp(a): import numpy as np; return np.uintp(a), np.array([a, '
 'a], dtype=np.uintp)')
        cxx_compiled = '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_ndarray_uintp.cpython-312-i386-linux-gnu.so'
        interface  = {'ndarray_uintp': [<class 'numpy.uint32'>]}
        modname    = 'test_ndarray_uintp'
        name       = 'ndarray_uintp'
        params     = (np.uint32(5),)
        prelude    = None
        python_ref = (np.uint32(5), array([5, 5], dtype=uint32))
        pythran_res = (np.uint32(5), array([5, 5], dtype=uint32))
        run_pythran_args = ('test_ndarray_uintp',
 '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_ndarray_uintp.cpython-312-i386-linux-gnu.so',
 ('ndarray_uintp', (np.uint32(5),)),
 None,
 False)
        self       = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>
        thread_count = 1
pythran/tests/__init__.py:118: in assertAlmostEqual
    self.assertAlmostEqual(ref_v, res_v)
        ref        = (np.uint32(5), array([5, 5], dtype=uint32))
        ref_v      = np.uint32(5)
        res        = (np.uint32(5), array([5, 5], dtype=uint32))
        res_v      = np.uint32(5)
        self       = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>
pythran/tests/__init__.py:114: in assertAlmostEqual
    self.check_type(ref, res)
        ref        = np.uint32(5)
        res        = np.uint32(5)
        self       = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>
ref = np.uint32(5), res = np.uint32(5)

    def check_type(self, ref, res):
        """ Check if type between reference and result match. """
        print("Type of Pythran res : ", type(res))
        print("Type of Python ref : ", type(ref))
    
        # cope with subtle situation under Windows where numpy.uint32 have same
        # name but different value
        if type(res).__name__ == type(ref).__name__:
            return
    
        if isinstance(ref, ndarray):
            # res can be an ndarray of dim 0 because of isneginf call
            if ref.ndim == 0 and (not isinstance(res, ndarray)
                                  or res.ndim != 0):
                self.check_type(ref.item(0), res)
            else:
                self.assertIsInstance(res, type(ref))
        elif isinstance(ref, (float, float64)):
            self.assertIsInstance(res, (float, float64))
        elif isinstance(ref, (complex, complex128)):
            self.assertIsInstance(res, (complex, complex128))
        elif isinstance(ref, (bool, bool_)):
            self.assertIsInstance(res, (bool, bool_))
        elif hasattr(ref, 'dtype'):
            if hasattr(res, 'dtype'):
                self.assertEqual(ref.dtype.itemsize, res.dtype.itemsize)
>               self.assertEqual(ref.dtype.type(-1), res.dtype.type(-1))
E               OverflowError: Python integer -1 out of bounds for uint32

ref        = np.uint32(5)
res        = np.uint32(5)
self       = <pythran.tests.test_ndarray.TestNdarray testMethod=test_ndarray_uintp>

pythran/tests/__init__.py:102: OverflowError
----------------------------- Captured stdout call -----------------------------
Python result:  (np.uint32(5), array([5, 5], dtype=uint32))
Pythran result:  (np.uint32(5), array([5, 5], dtype=uint32))
Type of Pythran res :  <class 'tuple'>
Type of Python ref :  <class 'tuple'>
Type of Pythran res :  <class 'numpy.uintc'>
Type of Python ref :  <class 'numpy.uint32'>
----------------------------- Captured stderr call -----------------------------
INFO: Generated module: test_ndarray_uintp
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_ndarray_uintp.cpython-312-i386-linux-gnu.so
------------------------------ Captured log call -------------------------------
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_ndarray_uintp' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpoyt4gwdm/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpimr8co86.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpoyt4gwdm/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpimr8co86.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpoyt4gwdm/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpimr8co86.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpqpszsjwo/test_ndarray_uintp.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_ndarray_uintp
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_ndarray_uintp.cpython-312-i386-linux-gnu.so
_ TestNumpyUFuncUnary.test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float _
[gw25] linux -- Python 3.12.9 /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/bin/python3.12

self = <pythran.tests.test_numpy_ufunc_unary.TestNumpyUFuncUnary testMethod=test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float>

>   ???

self       = <pythran.tests.test_numpy_ufunc_unary.TestNumpyUFuncUnary testMethod=test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float>

<string>:1: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:324: in run_test
    self.assertAlmostEqual(python_ref, pythran_res)
        check_exception = False
        code       = ('\n'
 'def numpy_ufunc_unary_numpy_uint32_scalar_float(a):\n'
 '    from numpy import uint32\n'
 '    return uint32(a)\n')
        cxx_compiled = '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_numpy_ufunc_unary_numpy_uint32_scalar_float.cpython-312-i386-linux-gnu.so'
        interface  = {'numpy_ufunc_unary_numpy_uint32_scalar_float': [<class 'float'>]}
        modname    = 'test_numpy_ufunc_unary_numpy_uint32_scalar_float'
        name       = 'numpy_ufunc_unary_numpy_uint32_scalar_float'
        params     = (0.5,)
        prelude    = None
        python_ref = np.uint32(0)
        pythran_res = np.uint32(0)
        run_pythran_args = ('test_numpy_ufunc_unary_numpy_uint32_scalar_float',
 '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_numpy_ufunc_unary_numpy_uint32_scalar_float.cpython-312-i386-linux-gnu.so',
 ('numpy_ufunc_unary_numpy_uint32_scalar_float', (0.5,)),
 None,
 False)
        self       = <pythran.tests.test_numpy_ufunc_unary.TestNumpyUFuncUnary testMethod=test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float>
        thread_count = 1
pythran/tests/__init__.py:114: in assertAlmostEqual
    self.check_type(ref, res)
        ref        = np.uint32(0)
        res        = np.uint32(0)
        self       = <pythran.tests.test_numpy_ufunc_unary.TestNumpyUFuncUnary testMethod=test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pythran.tests.test_numpy_ufunc_unary.TestNumpyUFuncUnary testMethod=test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float>
ref = np.uint32(0), res = np.uint32(0)

    def check_type(self, ref, res):
        """ Check if type between reference and result match. """
        print("Type of Pythran res : ", type(res))
        print("Type of Python ref : ", type(ref))
    
        # cope with subtle situation under Windows where numpy.uint32 have same
        # name but different value
        if type(res).__name__ == type(ref).__name__:
            return
    
        if isinstance(ref, ndarray):
            # res can be an ndarray of dim 0 because of isneginf call
            if ref.ndim == 0 and (not isinstance(res, ndarray)
                                  or res.ndim != 0):
                self.check_type(ref.item(0), res)
            else:
                self.assertIsInstance(res, type(ref))
        elif isinstance(ref, (float, float64)):
            self.assertIsInstance(res, (float, float64))
        elif isinstance(ref, (complex, complex128)):
            self.assertIsInstance(res, (complex, complex128))
        elif isinstance(ref, (bool, bool_)):
            self.assertIsInstance(res, (bool, bool_))
        elif hasattr(ref, 'dtype'):
            if hasattr(res, 'dtype'):
                self.assertEqual(ref.dtype.itemsize, res.dtype.itemsize)
>               self.assertEqual(ref.dtype.type(-1), res.dtype.type(-1))
E               OverflowError: Python integer -1 out of bounds for uint32

ref        = np.uint32(0)
res        = np.uint32(0)
self       = <pythran.tests.test_numpy_ufunc_unary.TestNumpyUFuncUnary testMethod=test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float>

pythran/tests/__init__.py:102: OverflowError
----------------------------- Captured stdout call -----------------------------
Python result:  0
Pythran result:  0
Type of Pythran res :  <class 'numpy.uintc'>
Type of Python ref :  <class 'numpy.uint32'>
----------------------------- Captured stderr call -----------------------------
INFO: Generated module: test_numpy_ufunc_unary_numpy_uint32_scalar_float
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_numpy_ufunc_unary_numpy_uint32_scalar_float.cpython-312-i386-linux-gnu.so
------------------------------ Captured log call -------------------------------
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_numpy_ufunc_unary_numpy_uint32_scalar_float' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpv144pdrs/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/numpy/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp8khw069r.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpv144pdrs/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp8khw069r.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORTIFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpv144pdrs/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmp8khw069r.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpm7aa1jff/test_numpy_ufunc_unary_numpy_uint32_scalar_float.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_numpy_ufunc_unary_numpy_uint32_scalar_float
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_numpy_ufunc_unary_numpy_uint32_scalar_float.cpython-312-i386-linux-gnu.so
```

Note that I've applied it on top of 0.17.0, in case that could be the problem.

Looks like it made things worse:

```pytb
FAILED pythran/tests/test_itertools.py::TestItertools::test_imap_over_array - TypeError: 'numpy.dtypes.Int32DType' object is not callab
le
FAILED pythran/tests/test_itertools.py::TestItertools::test_imap_over_several_arrays - TypeError: 'numpy.dtypes.Int32DType' object is n
ot callable
FAILED pythran/tests/test_conversion.py::TestConversion::test_builtin_type9 - TypeError: 'numpy.dtypes.Int32DType' object is not callab
le
FAILED pythran/tests/test_numpy_ufunc_unary.py::TestNumpyUFuncUnary::test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_int32_scalar_float 
- TypeError: 'numpy.dtypes.Int32DType' object is not callable
FAILED pythran/tests/test_ndarray.py::TestNdarray::test_ndarray_intp - TypeError: 'numpy.dtypes.Int32DType' object is not callable
FAILED pythran/tests/test_ndarray.py::TestNdarray::test_ndarray_uintp - TypeError: 'numpy.dtypes.UInt32DType' object is not callable
FAILED pythran/tests/test_numpy_ufunc_unary.py::TestNumpyUFuncUnary::test_numpy_ufunc_unary_numpy_ufunc_unary_numpy_uint32_scalar_float - TypeError: 'numpy.dtypes.UInt32DType' object is not callable
FAILED pythran/tests/test_cases.py::TestCases::test_collatz_run0 - TypeError: 'numpy.dtypes.Int32DType' object is not callable
FAILED pythran/tests/test_scipy.py::TestScipy::test_hausdorff_run0 - TypeError: 'numpy.dtypes.Int32DType' object is not callable
```

Example traceback:

```pytb
______________________ TestItertools.test_imap_over_array ______________________
[gw7] linux -- Python 3.12.9 /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/bin/python3.12

self = <pythran.tests.test_itertools.TestItertools testMethod=test_imap_over_array>

    def test_imap_over_array(self):
>       self.run_test("def imap_over_array(l):"
                      "  from numpy import arange ;"
                      "  t = tuple(map(lambda x: 1, (l,l))) ;"
                      "  return arange(10).reshape((5,2))[t]",
                      3,
                      imap_over_array=[int])

self       = <pythran.tests.test_itertools.TestItertools testMethod=test_imap_over_array>

pythran/tests/test_itertools.py:193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:329: in run_test
    self.assertAlmostEqual(python_ref, pythran_res)
        check_exception = False
        code       = ('def imap_over_array(l):  from numpy import arange ;  t = tuple(map(lambda x: '
 '1, (l,l))) ;  return arange(10).reshape((5,2))[t]')
        cxx_compiled = '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_imap_over_array.cpython-312-i386-linux-g
nu.so'
        interface  = {'imap_over_array': [<class 'int'>]}
        modname    = 'test_imap_over_array'
        name       = 'imap_over_array'
        params     = (3,)
        prelude    = None
        python_ref = np.int32(3)
        pythran_res = np.int32(3)
        run_pythran_args = ('test_imap_over_array',
 '/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_imap_over_array.cpython-312-i386-linux-gnu.so',
 ('imap_over_array', (3,)),
 None,
 False)
        self       = <pythran.tests.test_itertools.TestItertools testMethod=test_imap_over_array>
        thread_count = 1
pythran/tests/__init__.py:119: in assertAlmostEqual
    self.check_type(ref, res)
        ref        = np.int32(3)
        res        = np.int32(3)
        self       = <pythran.tests.test_itertools.TestItertools testMethod=test_imap_over_array>
pythran/tests/__init__.py:107: in check_type
    self.assertEqual(issigned(ref.dtype), issigned(res.dtype.type))
        issigned   = <function TestEnv.check_type.<locals>.issigned at 0xea98a258>
        ref        = np.int32(3)
        res        = np.int32(3)
        self       = <pythran.tests.test_itertools.TestItertools testMethod=test_imap_over_array>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

ty = dtype('int32')

    def issigned(ty):
        try:
>           return ty(-1) == -1
E           TypeError: 'numpy.dtypes.Int32DType' object is not callable

ty         = dtype('int32')

pythran/tests/__init__.py:103: TypeError
----------------------------- Captured stdout call -----------------------------
Python result:  3
Pythran result:  3
Type of Pythran res :  <class 'numpy.intc'>
Type of Python ref :  <class 'numpy.int32'>
----------------------------- Captured stderr call -----------------------------
INFO: Generated module: test_imap_over_array
INFO: Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_imap_over_array.cpython-312-i386-linux-gnu.so
------------------------------ Captured log call -------------------------------
INFO     root:dist.py:999 running build_ext
INFO     root:build_ext.py:535 building 'test_imap_over_array' extension
INFO     root:dir_util.py:58 creating /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmptdku37zp/var/tmp/portage/dev-python/pythran
-0.17.0-r1/temp
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 
-DPYTHRAN_BLAS_NONE -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/pythran -I/usr/lib/python3.12/site-packages/num
py/_core/include -I/var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0-python3_12/install/usr/include -I/usr/include/python3.12 -c /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpm4ua1ntq.cpp -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpt
dku37zp/var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmpm4ua1ntq.o -O2 -std=c++11 -fno-math-errno -fvisibility=hidden -fno-wrapv -
Wno-unused-function -Wno-int-in-bool-context -Wno-unknown-warning-option -O2 -march=i686 -pipe -msse2 -mfpmath=sse -O0 -UNDEBUG -U_FORT
IFY_SOURCE
INFO     root:spawn.py:60 i686-pc-linux-gnu-g++ -O2 -march=i686 -pipe -msse2 -mfpmath=sse -shared -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-r
elative-relocs -O2 -march=i686 -pipe -msse2 -mfpmath=sse /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmptdku37zp/var/tmp/portage
/dev-python/pythran-0.17.0-r1/temp/tmpm4ua1ntq.o -L/usr/lib -lcblas -o /var/tmp/portage/dev-python/pythran-0.17.0-r1/temp/tmppx_oo4x9/t
est_imap_over_array.cpython-312-i386-linux-gnu.so -fvisibility=hidden -Wl,-strip-all -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs
INFO     pythran:toolchain.py:388 Generated module: test_imap_over_array
INFO     pythran:toolchain.py:389 Output: /var/tmp/portage/dev-python/pythran-0.17.0-r1/work/pythran-0.17.0/test_imap_over_array.cpytho
n-312-i386-linux-gnu.so

```

--- a/pythran/tests/__init__.py
+++ b/pythran/tests/__init__.py
@@ -98,8 +98,13 @@ class TestEnv(unittest.TestCase):
             self.assertIsInstance(res, (bool, bool_))
         elif hasattr(ref, 'dtype'):
             if hasattr(res, 'dtype'):
+                def issigned(ty):
+                    try:
+                        return ty(-1) == -1
+                    except OverflowError:
+                        return False
                 self.assertEqual(ref.dtype.itemsize, res.dtype.itemsize)
-                self.assertEqual(ref.dtype.type(-1), res.dtype.type(-1))
+                self.assertEqual(issigned(ref.dtype.type), issigned(res.dtype.type))
             else:
                 self.assertIsInstance(res, int)
         else:

